ambry-router/src/test/java/com.github.ambry.router/PutManagerTest.java

```diff
@@ -614,23 +617,25 @@ private CountDownLatch submitPut()
       throws Exception {
     final CountDownLatch doneLatch = new CountDownLatch(requestAndResultsList.size());
     // This check is here for certain tests (like testConcurrentPuts) that require using the same router.
-    if (router == null || !router.isOpen()) {
+    if (instantiateNewRouterForPuts) {
```

Comment:
quick question before I merge this and it might be a bigger change so we can merge this anyway if that is the case.

Why should we ever need to instantiate a new `Router` unless we induced a close for testing purposes? In that case, those tests alone can start a new `Router` because they know they would have caused it to close. Otherwise, in a normal scenario, we would never need to create a new instance. Am I missing something?